[BUG] GCC 13 build warning

```shell
cmake . -DENABLE_BONDING=ON
make
```

produces the following warning with GCC 13.2 (Ubuntu 24.04)

```shell
[  1%] Building CXX object CMakeFiles/srt_virtual.dir/srtcore/group.cpp.o
In file included from /usr/include/c++/13/string:51,
                 from /usr/include/c++/13/bits/locale_classes.h:40,
                 from /usr/include/c++/13/bits/ios_base.h:41,
                 from /usr/include/c++/13/streambuf:43,
                 from /usr/include/c++/13/bits/streambuf_iterator.h:35,
                 from /usr/include/c++/13/iterator:66,
                 from /srt/srtcore/group.cpp:3:
In static member function ‘static _Up* std::__copy_move<_IsMove, true, std::random_access_iterator_tag>::__copy_m(_Tp*, _Tp*, _Up*) [with _Tp = unsigned char; _Up = unsigned char; bool _IsMove = false]’,
    inlined from ‘_OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = false; _II = unsigned char*; _OI = unsigned char*]’ at /usr/include/c++/13/bits/stl_algobase.h:506:30,
    inlined from ‘_OI std::__copy_move_a1(_II, _II, _OI) [with bool _IsMove = false; _II = unsigned char*; _OI = unsigned char*]’ at /usr/include/c++/13/bits/stl_algobase.h:533:42,
    inlined from ‘_OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = false; _II = unsigned char*; _OI = __gnu_cxx::__normal_iterator<unsigned char*, vector<unsigned char> >]’ at /usr/include/c++/13/bits/stl_algobase.h:540:31,
    inlined from ‘_OI std::copy(_II, _II, _OI) [with _II = unsigned char*; _OI = __gnu_cxx::__normal_iterator<unsigned char*, vector<unsigned char> >]’ at /usr/include/c++/13/bits/stl_algobase.h:633:7,
    inlined from ‘srt::CUDTGroup::ConfigItem::ConfigItem(SRT_SOCKOPT, const void*, int)’ at /srt/srtcore/group.h:88:22,
    inlined from ‘void srt::importOption(std::vector<CUDTGroup::ConfigItem>&, SRT_SOCKOPT, const ValueType&) [with ValueType = bool]’ at /srt/srtcore/group.cpp:458:41:
/usr/include/c++/13/bits/stl_algobase.h:437:30: warning: ‘void* __builtin_memcpy(void*, const void*, long unsigned int)’ forming offset 1 is out of the bounds [0, 1] of object ‘opt’ with type ‘bool’ [-Warray-bounds=]
  437 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/srt/srtcore/group.cpp: In function ‘void srt::importOption(std::vector<CUDTGroup::ConfigItem>&, SRT_SOCKOPT, const ValueType&) [with ValueType = bool]’:
/srt/srtcore/group.cpp:452:15: note: ‘opt’ declared here
  452 |     ValueType opt              = field;
      |               ^~~
[ 70%] Built target srt_virtual
```

--- a/scripts/release-notes/generate_release_notes.py
+++ b/scripts/release-notes/generate_release_notes.py
@@ -21,7 +21,7 @@ def define_area(msg):
         if msg.startswith(f'[{area}] '):
             return area
 
-    return np.NaN
+    return np.nan
 
 
 def delete_prefix(msg):

--- a/srtcore/group.cpp
+++ b/srtcore/group.cpp
@@ -447,11 +447,13 @@ static bool operator!=(const struct linger& l1, const struct linger& l2)
 template <class ValueType>
 static void importOption(vector<CUDTGroup::ConfigItem>& storage, SRT_SOCKOPT optname, const ValueType& field)
 {
-    ValueType default_opt      = ValueType();
-    int       default_opt_size = sizeof(ValueType);
-    ValueType opt              = field;
-    if (!getOptDefault(optname, (&default_opt), (default_opt_size)) || default_opt != opt)
+    ValueType default_opt             = ValueType();
+    static const int default_opt_size = sizeof(ValueType);
+    ValueType opt                     = field;
+    int opt_size                      = default_opt_size;
+    if (!getOptDefault(optname, (&default_opt), (opt_size)) || default_opt != opt)
     {
+        SRT_ASSERT(opt_size == default_opt_size);
         // Store the option when:
         // - no default for this option is found
         // - the option value retrieved from the field is different than default

--- a/srtcore/group.cpp
+++ b/srtcore/group.cpp
@@ -517,7 +517,11 @@ void CUDTGroup::deriveSettings(CUDT* u)
     IM(SRTO_UDP_RCVBUF, iUDPRcvBufSize);
     // SRTO_RENDEZVOUS: impossible to have it set on a listener socket.
     // SRTO_SNDTIMEO/RCVTIMEO: groupwise setting
-    IM(SRTO_CONNTIMEO, tdConnTimeOut);
+
+    // SRTO_CONNTIMEO requires a special handling, because API stores the value in integer milliseconds,
+    // but the type of the variable is srt::sync::duration.
+    importOption(m_config, SRTO_CONNTIMEO, (int) count_milliseconds(u->m_config.tdConnTimeOut));
+
     IM(SRTO_DRIFTTRACER, bDriftTracer);
     // Reuseaddr: true by default and should only be true.
     IM(SRTO_MAXBW, llMaxBW);
@@ -530,7 +534,9 @@ void CUDTGroup::deriveSettings(CUDT* u)
     IM(SRTO_RCVLATENCY, iRcvLatency);
     IM(SRTO_PEERLATENCY, iPeerLatency);
     IM(SRTO_SNDDROPDELAY, iSndDropDelay);
-    IM(SRTO_PAYLOADSIZE, zExpPayloadSize);
+    // Special handling of SRTO_PAYLOADSIZE becuase API stores the value as int32_t,
+    // while the config structure stores it as size_t.
+    importOption(m_config, SRTO_PAYLOADSIZE, (int)u->m_config.zExpPayloadSize);
     IMF(SRTO_TLPKTDROP, m_bTLPktDrop);
 
     importOption(m_config, SRTO_STREAMID, u->m_config.sStreamName.str());
@@ -544,7 +550,7 @@ void CUDTGroup::deriveSettings(CUDT* u)
 
     importOption(m_config, SRTO_PACKETFILTER, u->m_config.sPacketFilterConfig.str());
 
-    importOption(m_config, SRTO_PBKEYLEN, u->m_pCryptoControl->KeyLen());
+    importOption(m_config, SRTO_PBKEYLEN, (int) u->m_pCryptoControl->KeyLen());
 
     // Passphrase is empty by default. Decipher the passphrase and
     // store as passphrase option
@@ -824,7 +830,7 @@ void CUDTGroup::getOpt(SRT_SOCKOPT optname, void* pw_optval, int& w_optlen)
     if (w_optlen < int(i->value.size()))
         throw CUDTException(MJ_NOTSUP, MN_XSIZE, 0);
 
-    w_optlen = i->value.size();
+    w_optlen = (int) i->value.size();
     memcpy((pw_optval), &i->value[0], i->value.size());
 }
 

